src/vulkan/transfer_image.cc

```diff
@@ -182,15 +190,23 @@ Result TransferImage::CreateVkImageView() {
   return {};
 }
 
-VkBufferImageCopy TransferImage::CreateBufferImageCopy(uint32_t mip_level) {
+VkBufferImageCopy TransferImage::CreateBufferImageCopy(
+    VkImageAspectFlags aspect,
+    uint32_t mip_level) {
   VkBufferImageCopy copy_region = VkBufferImageCopy();
-  copy_region.bufferOffset = 0;
+  if (aspect == VK_IMAGE_ASPECT_STENCIL_BIT) {
+    // Store stencil data at the end of the buffer after depth data.
+    copy_region.bufferOffset =
```

Comment:
Does the buffers size already account for depth + stencil?

Reply:
It does. The size is queried from the driver.